recipes/opentracing-cpp/all/test_package/CMakeLists.txt

```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package CXX)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+find_package(OpenTracing REQUIRED CONFIG)
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} PRIVATE OpenTracing::opentracing-static)
```

Comment:
`OpenTracing::opentracing-static` won't be available when building a shared opentracing library.
You need to do
`cmake.definitions["OPENTRACING_SHARED"] = self.options["opentracing-cpp"].shared` in `test_package/conanfile.py`.
And link to the appropriate target in the test_package's `CMakeLists.txt`.

Reply:
Right, I'll add the if statement for the target.
About the `OPENTRACING_SHARED` variable, it's not defined on the original library so I don't think we should define it.

Reply:
I meant creating a new cmake `option` in test_package alone